src/Entity/ItemId.php

```diff
@@ -23,8 +24,9 @@ class ItemId extends EntityId implements Int32EntityId {
 	 * @throws InvalidArgumentException
 	 */
 	public function __construct( $idSerialization ) {
-		$this->assertValidIdFormat( $idSerialization );
-		$this->serialization = strtoupper( $idSerialization );
+		parent::__construct( self::upcaseLocalId( $idSerialization ) );
```

Comment:
> I suggest to rearrange the code in a way that you can reuse the split method, do a trivial strtoupper( $array[2] ) here in the subclass and then pass the array to the base class.

I don't understand the last bit here. Where would I pass an array to the base class and what would the base class do with it?